app/src/main/scala/org/alephium/explorer/persistence/schema/CustomSetParameter.scala

```diff
@@ -0,0 +1,143 @@
+    override def apply(input: BlockEntry.Hash, params: PositionedParameters): Unit =
+      params setBytes input.value.bytes.toArray
+  }
+
+  implicit object GroupIndexSetParameter extends SetParameter[GroupIndex] {
+    override def apply(input: GroupIndex, params: PositionedParameters): Unit =
+      params setInt input.value
+  }
+
+  implicit object ExplorerHashSetParameter extends SetParameter[explorer.Hash] {
+    override def apply(input: explorer.Hash, params: PositionedParameters): Unit =
+      params setBytes input.bytes.toArray
+  }
+
+  implicit object U256SetParameter extends SetParameter[U256] {
+    override def apply(input: U256, params: PositionedParameters): Unit =
+      params setBigDecimal BigDecimal(input.toBigInt)
+  }
+
+  implicit object AddressSetParameter extends SetParameter[Address] {
+    override def apply(input: Address, params: PositionedParameters): Unit =
+      params setString input.value
+  }
+
+  implicit object ByteStringSetParameter extends SetParameter[ByteString] {
+    override def apply(input: ByteString, params: PositionedParameters): Unit =
+      params setBytes input.toArray
+  }
+
+  implicit object BigIntegerSetParameter extends SetParameter[BigInteger] {
+    override def apply(input: BigInteger, params: PositionedParameters): Unit =
+      params setBigDecimal BigDecimal(input)
+  }
+
+  implicit object HeightSetParameter extends SetParameter[Height] {
+    override def apply(input: Height, params: PositionedParameters): Unit =
+      params setInt input.value
+  }
+
+  implicit object TransactionHashSetParameter extends SetParameter[Transaction.Hash] {
+    override def apply(input: Transaction.Hash, params: PositionedParameters): Unit =
+      params setBytes input.value.bytes.toArray
+  }
+
+  implicit object BlockEntryHashOptionSetParameter extends SetParameter[Option[BlockEntry.Hash]] {
+
+    /** {{{Params.setBytesOption(input.map(_.value.bytes.toArray[Byte]))}}} sets the value
+      * to java.lang.Object instead of null which fails and requires casting at SQL level.
+      *
+      * ERROR: org.postgresql.util.PSQLException: ERROR: column "***" is of type bytea but expression is of type oid
+      * Hint: You will need to rewrite or cast the expression.
+      *
+      * To keep this simple `null` is used and which set the column value as expected.
+      */
+    override def apply(input: Option[BlockEntry.Hash], params: PositionedParameters): Unit =
+      input match {
+        case Some(value) =>
+          params setBytes value.value.bytes.toArray
+
+        case None =>
+          //scalastyle:off null
+          params setBytes null
+        //scalastyle:on null
+      }
+  }
+
+  implicit object TimeStampSetParameter extends SetParameter[TimeStamp] {
+    // FIXME: There is an issue with timestamps returning incorrect values when queried.
+    //        The issue is reported and known. The code here will be determined by the
+    //        outcome of that other PR.
```

Comment:
fixed by https://github.com/alephium/explorer-backend/pull/153 :) let's not forget to fix the below lines to use `Long` once it's merged.